### PR TITLE
MVJ-550 override_receivable_type in Rent (UI)

### PR DESCRIPTION
**Pull request by juho-kettunen-nc**
This implements the UI part of override receivable type, where it is a property of **Rent**, instead of ContractRent as in https://github.com/City-of-Helsinki/mvj-ui/pull/544

### CODE REVIEW

**Comment by juho-kettunen-nc** [comment]
Still requires customer blessing to use this Rent version instead of the ContractRent, but this is the best version of the logic we have right now.

File: src/leases/helpers.ts
Comment (henrinie-nc) [inline src/leases/helpers.ts:2963]: If serviceunit has a type, the type of the argument here could be ServiceUnit["id"], which would end up with the same type number, but maybe providing more context

**Review by henrinie-nc — APPROVED** [review]
This seems usable. Improvement might be to add an attribute to the service unit model to dictate whether to use override field for this service unit, so we would not need to hard code lists of ID's and define code with service unit names.

Check my comments on the changes before merging.

File: src/leases/helpers.ts
Comment (juho-kettunen-nc) [reply src/leases/helpers.ts:2963]: Ah, didn't know that's possible

File: src/leases/components/LeasePage.tsx
Comment (juho-kettunen-nc) [reply src/leases/components/LeasePage.tsx:269]: No, because, these receivable types are only related to the lease's receivable type. The set of available receivable types should not change depending on the active service unit.

File: src/leases/components/leaseSections/rent/BasicInfoEdit.tsx
Comment (juho-kettunen-nc) [reply src/leases/components/leaseSections/rent/BasicInfoEdit.tsx:599]: Yes

**Comment by juho-kettunen-nc** [comment]
Added new field `use_rent_override_receivable_type` to the ServiceUnit model, and adjusted logic accordingly.
- API https://github.com/City-of-Helsinki/mvj/pull/772/commits/1277eed1f0152ecfb9f1f422d6a8a3adfeaa8355
- UI https://github.com/City-of-Helsinki/mvj-ui/pull/545/commits/fb1f3f31c8652e6b3a050104ed07587d06e47bf3

Input validation and tests in https://github.com/City-of-Helsinki/mvj/pull/772/commits/9e4c04186fca606be7cff99c9d3bdeda5e26edac

File: src/leases/helpers.ts
Comment (juho-kettunen-nc) [reply src/leases/helpers.ts]: Not relevant anymore because the function was removed

File: src/leases/components/leaseSections/rent/BasicInfoEdit.tsx
Comment (juho-kettunen-nc) [reply src/leases/components/leaseSections/rent/BasicInfoEdit.tsx]: Added in 96e3d70d93757cbcfcc625aa887d0e8d76c6d515

**Comment by henrinie-nc** [comment]
> Added new field `use_rent_override_receivable_type` to the ServiceUnit model, and adjusted logic accordingly.
> 
>     * API [City-of-Helsinki/mvj@1277eed](https://github.com/City-of-Helsinki/mvj/commit/1277eed1f0152ecfb9f1f422d6a8a3adfeaa8355)
> 
>     * UI [fb1f3f3](https://github.com/City-of-Helsinki/mvj-ui/commit/fb1f3f31c8652e6b3a050104ed07587d06e47bf3)
> 
> 
> Input validation and tests in [City-of-Helsinki/mvj@9e4c041](https://github.com/City-of-Helsinki/mvj/commit/9e4c04186fca606be7cff99c9d3bdeda5e26edac)

Looks good for the UI part!

**Comment by juho-kettunen-nc** [comment]
Reminder to self: squash, or rebase locally

**Pull request merged**
